Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -992,22 +1036,41 @@ bool AXTextMarker::isInTextRun() const
 AXTextMarkerRange AXTextMarker::lineRange(LineRangeType type) const
 {
     if (!isValid())
-        return { { }, { } };
+        return { };
 
     if (type == LineRangeType::Current) {
         auto startMarker = atLineStart() ? *this : previousLineStart();
         auto endMarker = atLineEnd() ? *this : nextLineEnd();
 
+        return { WTFMove(startMarker), WTFMove(endMarker) };
+    } else if (type == LineRangeType::Left) {
+        // Move backwards off a line start (because this is a "left-line" request).
+        auto startMarker = atLineStart() ? findMarker(AXDirection::Previous) : *this;
+        if (!startMarker.atLineStart())
+            startMarker = startMarker.previousLineStart();
+
+        auto endMarker = startMarker.nextLineEnd();
+        return { WTFMove(startMarker), WTFMove(endMarker) };
+    } else {
+        ASSERT(type == LineRangeType::Right);
+
+        // Move forwards off a line end (because this a "right-line" request).
+        auto startMarker = atLineEnd() ? findMarker(AXDirection::Next) : *this;
+        if (!startMarker.atLineStart())
+            startMarker = startMarker.previousLineStart();
+
+        auto endMarker = startMarker.nextLineEnd();
         return { WTFMove(startMarker), WTFMove(endMarker) };
     }
-    // FIXME: The other types aren't implemented yet.
-    RELEASE_ASSERT_NOT_REACHED();
+
+    ASSERT_NOT_REACHED();
```

Comment:
We can probably remove this `ASSERT` now, since we have an `else` case as a catch-all.